src/pages/auth/login/index.tsx

```diff
@@ -1,7 +1,106 @@
+import {
+  getProviders,
+  signIn,
+  getSession,
+  getCsrfToken,
+  useSession
+} from 'next-auth/react';
+import { Logo, Input } from '../../../components/Global';
+import { useRouter } from 'next/router';
+import styles from '../../../styles/pages/login.module.scss';
 
-const LoginPage = () => {
-  return <div>Login Page</div>;
+export default function LoginPage({ providers, csrfToken }) {
+  const { data: session, status } = useSession();
+  const passwordMinLength = 8;
+  const { error } = useRouter().query;
+
+  console.log(session);
+  const loginWithGoogle = event => {
+    signIn('google');
+  };
+  return (
+    <>
+      <div className={styles.logoLogin}>
+        <Logo showMark={true} showType={true} />
+      </div>
+      <div className={styles.mainSec}>
+        <h2 className={styles.title}>Login</h2>
+        {error && <SignInError error={error} />}
+        <form method="post" action="/api/auth/callback/credentials">
+          <div className={styles.inputContainer}>
+            <input name="csrfToken" type="hidden" defaultValue={csrfToken} />
+            <Input type={'email'} id={'email'} required={true} placeholder={'Email'}/>
+            <Input type={'password'} id={'password'} required={true} minLength={passwordMinLength} placeholder={'Password'}/>
+          </div>
+          <p className={styles.link}>
+            <a href={'/auth/signup'}>Do not have an account?</a>
+          </p>
+          <div className={styles.container}>
+            <button type="submit" className={styles.button}> Sign in with Credentials</button>
+            <button
+                type={'button'}
+                className={styles.button}
+                onClick={() => loginWithGoogle(event)}
+            >
+              Sign in with Google
+            </button>
+          </div>
+        </form>
+      </div>
+    </>
+  );
+}
+
+const errors = {
+  Signin: 'Try signing with a different account.',
+  OAuthSignin: 'Try signing with a different account.',
+  OAuthCallback: 'Try signing with a different account.',
+  OAuthCreateAccount: 'Try signing with a different account.',
+  EmailCreateAccount: 'Try signing with a different account.',
+  Callback: 'Try signing with a different account.',
+  OAuthAccountNotLinked:
+    'To confirm your identity, sign in with the same account you used originally.',
+  EmailSignin: 'Check your email address.',
+  CredentialsSignin:
+    'Sign in failed. Check the details you provided are correct.',
+  default: 'Unable to sign in.'
 };
+const SignInError = ({ error }) => {
+  const errorMessage = error && (errors[error] ?? errors.default);
+  return <div>{errorMessage}</div>;
+};
+
+// This is the recommended way for Next.js 9.3 or newer
+export async function getServerSideProps(context) {
+  //const {req, res} = context;
+  const session = await getSession(context);
+  if (session) {
+    return {
+      redirect: {
+        destination: '/',
+        permanent: false
```

Comment:
how is this different from router.push('./), just asking 🤓

Reply:
Redirects do not affect client-side routing while router.push( ) handles client-side transitions. Since this is in the getServerSideProps function, redirect( ) suits the condition better.